cmd/snap-confine/context-support.c

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "config.h"
+#include "context-support.h"
+#include "utils.h"
+
+#include <string.h>
+#include <unistd.h>
+#include <errno.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+
+#define CONTEXTS_DIR "/var/lib/snapd/contexts"
+
+char *read_snap_context(const char *snap_name)
+{
+	char context_path[1000];
+	char *context_val = NULL;
+
+	if (snap_name == NULL) {
+		die("SNAP_NAME is not set");
+	}
+
+	must_snprintf(context_path, 1000, "%s/snap.%s", CONTEXTS_DIR,
+		      snap_name);
+
+	int fd = open(context_path, O_RDONLY);
+	if (fd < 0) {
+		error
+		    ("Cannot open context file %s, SNAP_CONTEXT will not be set: %s",
+		     context_path, strerror(errno));
```

Comment:
Use die or switch to errors, they handle strerror part internally.

If the design is to make the absence of context non-fatal then switch to errors and ensure the function has `nonfatal` in the name.